command.go

```diff
@@ -1418,6 +1418,13 @@ func (c *Command) LocalFlags() *flag.FlagSet {
 	addToLocal := func(f *flag.Flag) {
 		if c.lflags.Lookup(f.Name) == nil && c.parentsPflags.Lookup(f.Name) == nil {
 			c.lflags.AddFlag(f)
+
+			// If `f` has already been changed, re-add its value to the flag set,
+			// otherwise `c.lflags` has no record of the flags value.
+			if f.Changed {
+				f.Changed = false
```

Comment:
Why is it that you need to unset the `Changed` field? I thought the point was to just copy over the new value, I didn't expect to see this modified as well.